evil-commands.el

```diff
@@ -5168,6 +5168,21 @@ event that triggered the execution of this command."
     (push event unread-command-events)))
 (evil-declare-ignore-repeat 'evil-exit-visual-and-repeat)
 
+(evil-define-command evil-retab (tabstop)
+  "Convert all tabs to spaces or the other way around.
+Replace all sequences of white-space containing a <Tab> with new
+strings of white-space using the new TABSTOP value given.
+If you do not specify a new TABSTOP size or it is zero, Evil uses the
+current value of `tab-width'."
+  (interactive "<a>")
+  (let ((beg (if (use-region-p) (region-beginning) (point-min)))
+        (end (if (use-region-p) (region-end) (point-max)))
+        (retab (if indent-tabs-mode #'tabify #'untabify))
+        (tab-width (cond ((not tabstop) tab-width)
+                         ((equal tabstop "0") tab-width)
+                         (t (string-to-number tabstop)))))
```

Comment:
I think `string-to-number` is too permissive. e.g.
```elisp
(string-to-number "foo")
0
(string-to-number "7foo")
7
```
This  (before the let binding, even) would catch most things:
```elisp
(string-match-p "^[0-9]*$" tabstop)
```

Reply:
Good catch, fixed.